spec/ordersystem_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'ordersystem'
+
+describe OrderSystem do
+  let(:dish) { double :dish }
+  let(:option_one) { { dish: dish, price: 3 } }
+  let(:option_two) { { dish: dish, price: 5 } }
+  let(:takeaway) { TakeawayKitchen.new.menu }
+  
+  it 'returns the menu from TakeawayKitchen' do
+    expect(subject.menu).to eq takeaway
+  end
+
+  it 'initializes with an empty order' do
+    expect(subject.order).to be_empty
+  end
+
+  describe '#view_menu' do
+    it 'returns a hash' do
+      expect(subject.view_menu).to include({ dish: be_an(String), price: be_an(Integer) })
+    end
+  end
+
+  describe '#add_to_order' do
+    it { is_expected.to respond_to(:add_to_order).with(2).argument }
+
+    it 'increases order count' do
+      subject.menu << option_one
+      subject.add_to_order dish
+      expect(subject.order.count).to eq 1
```

Comment:
This test feels very stateful. You want to test behaviour instead. How would your user test that this method worked? Maybe they looked at the return, or maybe they had to call view menu, or something else to make sure their order had been updated. You want to test this the way the user of the class would test it.

Reply:
See the "testing state vs behaviour" week 2 practical.